apps/print/frontend/src/components/toolbar.tsx

```diff
@@ -26,15 +30,125 @@ function useCurrentDate(): Date {
   return currentDate;
 }
 
-function PrinterStatus({ status }: { status: PrinterStatusType }) {
+function BasePrinterStatus({
+  icon,
+  labelText,
+}: {
+  icon?: JSX.Element;
+  labelText?: string;
+}) {
   return (
-    <Row style={{ gap: '0.25rem', alignItems: 'center' }}>
+    <Row style={{ gap: '0.25rem', justifyContent: 'end' }}>
       <Icons.Print color="inverse" />
-      {!status.connected && <Icons.Warning color="inverseWarning" />}
+      {icon}
+      {labelText && <Caption>{labelText}</Caption>}
     </Row>
   );
 }
 
+function PrinterConnectionStatus({ connected }: { connected: boolean }) {
+  if (connected) {
+    return <BasePrinterStatus />;
+  }
+
+  return (
+    <BasePrinterStatus
+      icon={<Icons.Disabled color="danger" />}
+      labelText="Not Connected"
+    />
+  );
+}
+
+function PrinterStatus({ status }: { status: PrinterStatusType }) {
+  const { connected } = status;
+  if (!connected || !status.richStatus) {
+    return <PrinterConnectionStatus connected={connected} />;
+  }
+
+  const { richStatus } = status;
+
+  console.log(richStatus);
+
+  if (richStatus.state === 'stopped') {
+    if (
+      richStatus.stateReasons.find((reason) => reason === 'media-empty-error')
+    ) {
+      // No paper and paper tray open both return the same error, so we can't
+      // differentiate the error message
+      return (
+        <BasePrinterStatus
+          icon={<Icons.Danger color="danger" />}
+          labelText="No Paper"
+        />
+      );
+    }
+
+    if (
+      richStatus.stateReasons.find((reason) => reason === 'media-jam-error')
+    ) {
+      return (
+        <BasePrinterStatus
+          icon={<Icons.Danger color="danger" />}
+          labelText="Jammed"
+        />
+      );
+    }
+
+    // Printer output is obstructed
+    if (
+      richStatus.stateReasons.find(
+        (reason) =>
+          reason === 'spool-area-full' || reason === 'spool-area-full-report'
+      )
+    ) {
+      return (
+        <BasePrinterStatus
+          icon={<Icons.Danger color="danger" />}
+          labelText="Output Full"
```

Comment:
I'm new to the printer world but wondering if the term "Output Full" is intuitive? Though I'm not sure I can think of anything better. "Tray Full" comes to mind, but not sure that it's an improvement